packages/inference/src/providers/wavespeed-ai.ts

```diff
@@ -0,0 +1,178 @@
+import { InferenceOutputError } from "../lib/InferenceOutputError";
+import type { ImageToImageArgs } from "../tasks";
+import type { BodyParams, HeaderParams, RequestArgs, UrlParams } from "../types";
+import { delay } from "../utils/delay";
+import { omit } from "../utils/omit";
+import { base64FromBytes } from "../utils/base64FromBytes";
+import {
+	TaskProviderHelper,
+	TextToImageTaskHelper,
+	TextToVideoTaskHelper,
+	ImageToImageTaskHelper,
+} from "./providerHelper";
+
+const WAVESPEEDAI_API_BASE_URL = "https://api.wavespeed.ai";
+
+/**
+ * Common response structure for all WaveSpeed AI API responses
+ */
+interface WaveSpeedAICommonResponse<T> {
+	code: number;
+	message: string;
+	data: T;
+}
+
+/**
+ * Response structure for task status and results
+ */
+interface WaveSpeedAITaskResponse {
+	id: string;
+	model: string;
+	outputs: string[];
+	urls: {
+		get: string;
+	};
+	has_nsfw_contents: boolean[];
+	status: "created" | "processing" | "completed" | "failed";
+	created_at: string;
+	error: string;
+	executionTime: number;
+	timings: {
+		inference: number;
+	};
+}
+
+/**
+ * Response structure for initial task submission
+ */
+interface WaveSpeedAISubmitResponse {
+	id: string;
+	urls: {
+		get: string;
+	};
+}
+
+type WaveSpeedAIResponse<T = WaveSpeedAITaskResponse> = WaveSpeedAICommonResponse<T>;
+
+abstract class WavespeedAITask extends TaskProviderHelper {
+	private accessToken: string | undefined;
+
+	constructor(url?: string) {
+		super("wavespeed-ai", url || WAVESPEEDAI_API_BASE_URL);
+	}
+
+	makeRoute(params: UrlParams): string {
+		return `/api/v2/${params.model}`;
+	}
+	preparePayload(params: BodyParams): Record<string, unknown> {
+		const payload: Record<string, unknown> = {
+			...omit(params.args, ["inputs", "parameters"]),
+			...(params.args.parameters as Record<string, unknown>),
+			prompt: params.args.inputs,
+		};
+		// Add LoRA support if adapter is specified in the mapping
+		if (params.mapping?.adapter === "lora" && params.mapping.adapterWeightsPath) {
+			payload.loras = [
+				{
+					path: params.mapping.adapterWeightsPath,
```

Comment:
For reference, `adapterWeightsPath` is the path to the LoRA weights inside the associated HF repo
eg, for [nerijs/pixel-art-xl](https://huggingface.co/api/models/nerijs/pixel-art-xl?expand[]=inferenceProviderMapping), it will be 
```
"pixel-art-xl.safetensors"
```

Let's make sure that is indeed what your API is expecting when running LoRAs

Reply:
Here I see that fal is the endpoint that has been concatenated with hf.
Can I directly set the adapterWeightsPath to a lora http address? Or any other address.

Reply:
In the test cases, I conducted the test in this way. The `adapterWeightsPath` was directly passed over as an input parameter of lora.
```
"wavespeed-ai/flux-dev-lora": {
	hfModelId: "wavespeed-ai/flux-dev-lora",
	providerId: "wavespeed-ai/flux-dev-lora",
	status: "live",
	task: "text-to-image",
	adapter: "lora",
	adapterWeightsPath:
		"https://d32s1zkpjdc4b1.cloudfront.net/predictions/599f3739f5354afc8a76a12042736bfd/1.safetensors",
},
"wavespeed-ai/flux-dev-lora-ultra-fast": {
	hfModelId: "wavespeed-ai/flux-dev-lora-ultra-fast",
	providerId: "wavespeed-ai/flux-dev-lora-ultra-fast",
	status: "live",
	task: "text-to-image",
	adapter: "lora",
	adapterWeightsPath: "linoyts/yarn_art_Flux_LoRA",
},
```

in wavespeedai task is :
<img width="693" alt="image" src="https://github.com/user-attachments/assets/d5a51d07-b668-4b79-9cd7-30c6ca0e9bad" />
<img width="813" alt="image" src="https://github.com/user-attachments/assets/0827fcfd-4746-4def-bd82-1d527d9ecf1f" />

However, I'm not sure whether the input parameters submitted by hf to lora must be the abbreviation of the file path of the hf model and then concatenated with the hf address in the code. If it is this kind of specification, I can complete it in the format of fal

Reply:
I think your API can just take the hf model id as the loras path, right?
```suggestion
					path: params.mapping.hfModelId,,
```
As mentioned by @SBrandeis, this part depends on what your API is expecting as inputs when using LoRAs weights.

Reply:
Yes, you're correct. 
In the example, `linoyts/yarn_art_Flux_LoRA` is the lora model address of hf. We will automatically match and download the hf model。